### PR TITLE
Implementing DuckDuckGo Bangs frictionless access

### PR DESCRIPTION
**Pull request by DyeffersonAz**
"Bangs" is a very good tool present on DuckDuckGo to search on other websites. For more information, click [here](https://duckduckgo.com/bang).

People using Albert would need to use WebSearch extension _(prefixing the query with "duckduckgo" [Or something like that, as I don't remember what is the default trigger for that])_ to get to DuckDuckGo and only then use the bangs. With this module, You can use the bangs directly into the Albert's bar.

Maybe this needs to be polished in aesthetics or conventions (I don't think so), but it is now completely functional for me.

### CODE REVIEW

**Comment by bergercookie** [comment]
Do you get autocompletion for your search results?
How is it different to the ddgr-powered plugins of this repo:

- https://github.com/bergercookie/awesome-albert-plugins/blob/master/create_ddgr_plugins.py#L94=
- https://github.com/bergercookie/awesome-albert-plugins#ddgr-based-plugins=

For ddgr see https://github.com/jarun/ddgr

**Comment by DyeffersonAz** [comment]
In fact, it is, indeed, an extension that does less than those you just told me about. I didn't know about those existent tools. This PR just shows "the world" a solution I made for myself, since this functionality is used by me daily since March. It's a specific extension that does a specific thing that is not present in the official version of albert. I didn't know about "ddr" either. Thanks for sharing it!
Autocompletion isn't a thing for "bangs", is it?

**Comment by bergercookie** [comment]
>  Autocompletion isn't a thing happening in "bangs", is it?

No idea, I haven't used bangs that much 😅

**Review by bergercookie — APPROVED** [review]

**Comment by DyeffersonAz** [comment]
Hi, @bergercookie ! I was looking at your plugin at awesome-albert-plugins and I saw that you hard-coded the number of websites that DuckDuckGo can search directly by using an array. Why did you do that? If someone knows the bang shortcode, every website supported by DuckDuckGo bangs would work along with it.
Also, great work there. A ton of features! Congratulations!

**Comment by bergercookie** [comment]
> Why did you do that? If someone knows the bang shortcode, every website supported by DuckDuckGo bangs would work along with it.

That's not a bad shout, seems like `ddgr` which I'm using under the hood does support  bangs as well, so if the code is not a registered one (as the ones I've added to the python script and added symbols for, then we could fallback to the bang list - https://github.com/jarun/ddgr#examples

Feel free to send me a PR if you're interested :)

File: bangs/__init__.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    item.icon = iconPath
+
+    if bangName and searchQuery:
+        item.text = "Bang " + bangName + ' used to search for "' + searchQuery + '".'
+    else:
+        item.text = "Bang!"
+        item.subtext = "Search faster!"
+
+    item.actions = [
+        UrlAction(text="UrlAction", url="https://duckduckgo.com/?q=" + query.rawString)
```
[the remark this answers is not in the document]
Comment (idkCpp) [reply bangs/__init__.py:37]: Furthermore, wouldn't this need url encoding?

**Comment by ManuelSchneid3r** [comment]
Final I managed to have a somewhat stable 0.18 api please update

**Comment by bergercookie** [comment]
> stable 0.18 api

Nice!
Is this in master and in the debian packages ?
Any changelog as to what's required to update plugins from earlier versions?

**Comment by ManuelSchneid3r** [comment]
0.18 is not tagged yet. I expect a lot to break. Consider it a developer preview. I merged into master yesterday. I am going to write the docs today I guess.

**Comment by ManuelSchneid3r** [comment]
@DyeffersonAz this is the native plugin repo. Therefore I am going to close this one here.

Did you intend to send the PR here? https://github.com/albertlauncher/python

**Pull request closed without merging**